### DISCUSSION TITLE
Do not assign value to Patient ID attribute

### DISCUSSION THREAD

**Discussion by hackermd** [dcm4che/dcm4chee-arc-light#3796] [Q&A]
**Describe the bug**
The archive appears to automatically assign a value to the Patient ID attribute if none of the patient or study level attributes have a value. These are all type 2 attributes and so the values may be rightfully missing (e.g., after de-identification).

The archive should not just modify the data upon storage. This behavior is very problematic. For example, if one stores a set of images in the archive, then derives other objects from these images (e.g., an SR document), inheriting the patient and study level metadata from the source image, and finally downloads the derived object, then the metadata in the derived object will be different from the image that was stored in the first place. If one then tries to store both the original images and derived object in another archive, this archive will complain that the metadata is different and will refuse to store the data.

**Comment by gunterze** [comment]
The behavior is configurable by attribute

Name | Type | Description (LDAP Attribute)
-- | -- | --
Accept Missing Patient ID | string | Indicates if objects without Patient IDs shall be accepted and if  a Patient ID shall be created. Enumerated values: YES, NO or CREATE. (dcmAcceptMissingPatientID)

on [Archive Device](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/archiveDevice.html#dcmacceptmissingpatientid) and [Archive Network AE](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/archiveNetworkAE.html#dcmacceptmissingpatientid) level.

**Reply by hackermd** [reply]
It's nice that this is configurable. I would suggest, however, considering changing the default value. Modifying data by default is a very bad idea in my opinion, as illustrated by the provided example. The storage of data via the DICOMweb Store Transaction should not result in modification of the data. I consider this behavior a violation of the DICOM standard.
